source/adapters/level_zero/event.cpp

```diff
@@ -434,8 +456,22 @@ ur_result_t urEnqueueEventsWaitWithBarrierExt(
         *OutEvent ///< [in,out][optional] return an event object that identifies
                   ///< this particular command instance.
 ) {
-  return ur::level_zero::urEnqueueEventsWaitWithBarrier(
-      Queue, NumEventsInWaitList, EventWaitList, OutEvent);
+  bool InterruptBased =
+      EnqueueExtProp &&
+      (EnqueueExtProp->flags & UR_EXP_ENQUEUE_EXT_FLAG_LOW_POWER_EVENTS);
+  if (InterruptBased) {
+    // Create the event with interrupt-based properties
+    return static_cast<ur_result_t (*)(
+        ur_queue_handle_t, uint32_t, const ur_event_handle_t *,
+        ur_event_handle_t *, bool)>(EnqueueEventsWaitWithBarrier)(
+        Queue, NumEventsInWaitList, EventWaitList, OutEvent, true);
+  } else {
+    return static_cast<ur_result_t (*)(
+        ur_queue_handle_t, uint32_t, const ur_event_handle_t *,
+        ur_event_handle_t *, bool)>(EnqueueEventsWaitWithBarrier)(
+        Queue, NumEventsInWaitList, EventWaitList, OutEvent,
+        Queue ? Queue->InterruptBasedEventsEnabled : false);
```

Comment:
`Queue ? Queue->InterruptBasedEventsEnabled : false`
I don't believe Queue can be null here. So this should be simply `Queue->InterruptBasedEventsEnabled`.

Reply:
I agree, this is not needed, please remove and simplify.